### PR TITLE
feat: parse anti-abuse bond tags from kind-38385 info event

### LINKED ISSUE #594: Read & cache new anti-abuse-bond tags from the kind-38385 info event
## Context

The mostrod daemon publishes a **kind-38385 info event** with the node's runtime policy: `mostro_version`, `max_order_amount`, `expiration_seconds`, `fee`, `pow`, LND status, etc. The mobile client already reads this event and caches the values internally.

[mostro@ad1f74a](https://github.com/MostroP2P/mostro/commit/ad1f74a) ships **seven new tags** on the same event that advertise the [anti-abuse bond](https://github.com/MostroP2P/mostro/blob/main/docs/ANTI_ABUSE_BOND.md) policy of the connected node. These tags are critical for the [Anti-Abuse Bond Support](https://github.com/MostroP2P/mobile/issues/591) workstream — without them the app can't:

- show the user, before they enter a trade, whether a bond is required, how much it will cost, and under what conditions it can be slashed;
- render the **payout forfeit deadline** when the daemon asks the winning counterparty for a payout invoice. The \`Action::AddBondInvoice\` message deliberately carries **no human-readable text** (Mostro stays locale-agnostic). The client computes the deadline as \`slashed_at + bond_payout_claim_window_days * 86400\`, where:
  - \`slashed_at\` (Unix seconds, UTC) is shipped on the \`Action::AddBondInvoice\` message body in the new \`Payload::BondPayoutRequest { order: SmallOrder, slashed_at: i64 }\` variant (mostro-core **0.11.3**). It is the **fixed anchor**: re-shipped verbatim on every cadence retry, so a recipient who was offline for days still computes the correct deadline once the message lands.
  - \`bond_payout_claim_window_days\` is the kind-38385 info-event tag this issue is about.

  This split is deliberate — relying on message receipt time for \`slashed_at\` would silently drift the deadline into the future for offline recipients. Handling the \`BondPayoutRequest\` payload itself is part of the umbrella #591; **this** issue only needs to surface \`bond_payout_claim_window_days\` from the info event so the deadline computation has both inputs available.

## Purpose of this issue

**Just the reader/cache layer.** The UI consumption, confirmation dialogs, and trade-flow integration are tracked in #591. This issue is scoped to: read these tags from the kind-38385 event and store them internally **exactly like the other info-event tags already are** (same model, same cache, same refresh cadence — whatever pattern is already in place for \`mostro_version\` / \`fee\` / etc.).

## New tags

| Tag | Type | Values | When emitted |
|---|---|---|---|
| \`bond_enabled\` | bool | \`"true"\` \| \`"false"\` | **Always** — even when the feature is off (see "Disambiguation" below) |
| \`bond_apply_to\` | enum | \`"take"\` \| \`"make"\` \| \`"both"\` | Only when \`bond_enabled = true\` |
| \`bond_slash_on_waiting_timeout\` | bool | \`"true"\` \| \`"false"\` | Only when \`bond_enabled = true\` |
| \`bond_amount_pct\` | float | e.g. \`"0.01"\` (1% of order amount) | Only when \`bond_enabled = true\` |
| \`bond_base_amount_sats\` | integer | sats floor, e.g. \`"1000"\` | Only when \`bond_enabled = true\` |
| \`bond_slash_node_share_pct\` | float in \`[0.0, 1.0]\` | e.g. \`"0.5"\` | Only when \`bond_enabled = true\` |
| \`bond_payout_claim_window_days\` | integer | days, e.g. \`"15"\` | Only when \`bond_enabled = true\` |

Bond amount math: \`max(bond_amount_pct * order_amount_sats, bond_base_amount_sats)\`.

Slash split math: the winning counterparty receives \`(1 - bond_slash_node_share_pct) * amount_sats\`; the node retains the rest.

## Disambiguation: three states, not two

Older mostrod versions don't emit \`bond_enabled\` at all. So the client must distinguish:

1. **Tag absent** → daemon doesn't speak bond; treat as legacy / no bond.
2. **\`bond_enabled = "false"\`** → modern daemon, but the operator hasn't enabled the feature; no bond will be requested.
3. **\`bond_enabled = "true"\`** → bond is active; the other six tags will also be present.

This is the reason \`bond_enabled\` is the only one always emitted — please preserve that semantic when modelling the parsed state.

## References

- Daemon spec §13.1: <https://github.com/MostroP2P/mostro/blob/main/docs/ANTI_ABUSE_BOND.md#131-scope>
- Daemon tag emitter: \`src/nip33.rs::info_to_tags\`
- Umbrella mobile issue: #591

### PR DESCRIPTION
**Pull request by AndreaDiazCorreia**
Closes #594

## Summary

Adds the reader/cache layer for the seven new anti-abuse bond tags shipped on `mostrod`'s kind-38385 info event ([ad1f74a](https://github.com/MostroP2P/mostro/commit/ad1f74a)). Required by the upcoming Anti-Abuse Bond Support work in #591.

Scope is intentionally limited to parsing and caching. UI, trade-flow integration, and payout deadline computation land with #591.

## Three-state semantics

`bond_enabled` is the only bond tag emitted unconditionally on modern daemons. Modelled as a `BondPolicy` enum rather than `bool?` to prevent silently collapsing "tag absent" with `bond_enabled = "false"`:

- `unsupported` — tag absent (legacy daemon)
- `disabled` — `bond_enabled = "false"`
- `enabled` — `bond_enabled = "true"` (six other bond tags present)

## Changes

- `lib/features/mostro/mostro_instance.dart`: new `BondPolicy` and `BondApplyTo` enums, seven new fields on `MostroInstance`, matching getters on the `NostrEvent` extension with defensive parsing.
- `test/features/mostro/mostro_instance_test.dart`: unit tests covering the three policy states, enum parsing, and malformed values.

## Summary by CodeRabbit

* **New Features**
  * Instances can now advertise anti-abuse bond support, scope (take/make/both) and configurable bond parameters; parsing trims whitespace, treats missing/invalid values as unspecified, and makes payout-claim-window optional when not provided.

* **Tests**
  * Expanded test coverage for bond flag parsing, scope mapping, boolean/numeric validation and bounds, and related edge cases.

[Review Change Stack](https://app.coderabbit.ai/change-stack/MostroP2P/mobile/pull/601?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: lib/features/mostro/mostro_instance.dart [hunk 1]
```diff
@@ -103,5 +164,79 @@ extension MostroInstanceExtensions on NostrEvent {
...
+
+  BondApplyTo? get bondApplyTo {
+    final raw = _getOptionalTagValue('bond_apply_to');
+    switch (raw) {
+      case 'take':
+        return BondApplyTo.take;
+      case 'make':
+        return BondApplyTo.make;
+      case 'both':
+        return BondApplyTo.both;
```
[the remark this answers is not in the document]
Comment (AndreaDiazCorreia) [reply lib/features/mostro/mostro_instance.dart:200]: The current implementation on commit 2f2a51f2 already returns null for any value other than "true"/"false" (case-insensitive). See the bondSlashOnWaitingTimeout getter and the test bondSlashOnWaitingTimeout returns null for malformed values which exercises 'foo', 'yes', 'no', '1', '0', and ''. Could you re-check?

**Review by Catrya — APPROVED** [review]
tACK
